react/features/follow-me/middleware.js

```diff
@@ -0,0 +1,177 @@
+// @flow
+
+import { MiddlewareRegistry } from '../base/redux';
+import {
+    getLocalParticipant,
+    getParticipantById,
+    getPinnedParticipant,
+    pinParticipant
+} from '../base/participants';
+import { CONFERENCE_WILL_JOIN } from '../base/conference';
+import { setFilmstripVisible } from '../filmstrip';
+import { setTileView } from '../video-layout';
+
+import { FOLLOW_COMMAND } from './constants';
+
+const logger = require('jitsi-meet-logger').getLogger(__filename);
+
+declare var APP: Object;
+
+/**
+ * The timeout after which a follow-me command that has been received will be
+ * ignored if not consumed.
+ *
+ * @type {number} in seconds
+ * @private
+ */
+const _FOLLOW_ME_RECEIVED_TIMEOUT = 30;
+
+/**
+ * An instance of a timeout used as a workaround when attempting to pin a
+ * non-existent particapant, which may be caused by participant join information
+ * not being received yet.
+ *
+ * @type {TimeoutID}
+ */
+let nextOnStageTimeout;
```

Comment:
Generally speaking, we try to avoid globals in new / refactored code. Any reason why we cannot store this in the redux store?

Reply:
You mean you avoid module scoped variables? This is currently module scoped, right? We probably can store it in redux but why store non-app state in redux? The reason why I put it in the module scope: laziness and copying the previous implementation. I can put it in redux if you're sure you want it there.

Reply:
Nah, ok, leave it there.